City-Bureau/city-scrapers-spokane
# 🏗️ Build spider: Spokane Valley City Council

## What's this PR do?
Adds a spider to scrape Spokane Valley City Council meetings.

## Why are we doing this?
Requested by City Bureau's Spokane partners.

## Steps to manually test

1. Ensure the project is installed:
```
pipenv sync --dev
```
2. Activate the virtual env and enter the pipenv shell:
```
pipenv shell
```
3. Run the spider:
```
scrapy crawl sv_city_council -O test_output.csv
```
4. Monitor the output and ensure no errors are raised.

5. Inspect `test_output.csv` to ensure the data looks valid. Target website is [here](https://spokanevalley.granicus.com/ViewPublisher.php?view_id=3).

## Are there any smells or added technical debt to note?
- This spider intentionally ignores the target's `robots.txt` file in order to parse meeting data. That is not preferable but there appears to be a precedent within the city-scrapers project of doing this. Eg. [this spider](https://github.com/City-Bureau/city-scrapers/blob/f897d5191396ced70a84f1be0630c3e6d81a25fc/city_scrapers/spiders/il_elections.py) in `city-scrapers`. Many of its sister spiders do the same.
- This page is a bit lacking in detailed meeting information, but we're parsing what we can. Helpfully, the page includes unix timestamps for meeting start times. However, this required some nuanced handling in my `parse_start` method because city-scrapers-core doesn't expect datetimes that are timezone aware.
